src/ja/setting-up-for-tdd.md

```diff
@@ -1 +1,257 @@
+
+```
+import { mount } from '@vue/test-utils'
+import Greeting from '@/components/Greeting.vue'
+
+describe('Greeting.vue', () => {
+  it('renders a greeting', () => {
+
+  })
+})
+```
+
+TDD に用いられるシンタックスは様々ですが、 Jest においては `describe` と `it` が比較的多く使われているようです。 `describe` はテストがどのようなものなのか、概要を示すために記述します。今回のケースでは `Greeting.vue` という部分ですね。`it` は、それによって題名を与えられたテストが、どのような目的を果たすためにあるのか、ということを示します。コンポーネントの機能が増えた際には、`it` ブロックを増やしていけばいいわけです。
+
+There are differnet syntaxes used for TDD, we will use the commonly seen `describe` and `it` syntax that comes with Jest. `describe` generally outlines what the test is about, in this case `Greeting.vue`. `it` represents a single piece of responsility that the subject of the test should fulfil. As we add more features to the component, we add more `it` blocks.
+
+さて、コンポーンネントを `mount` を用いて render せねばなりません。この一般的な慣習として、これによって生じたコンポーネントを `wrapper` を呼ばれる変数に紐付けます。さらにその内容を画面に出力することにしましょう。これで正常に動作しているか確認できますね。
+
+Now we should render the component with `mount`. The standard practice it to assign the component to a variable called `wrapper`. We will also print the output, to make sure everything is running correctly:
+
+```js
+const wrapper = mount(Greeting)
+
+console.log(wrapper.html())
+```
+
+## テストを走らせる
+
+`yarn test:unit`　とターミナルに打ち込んで実行することでテストを走らせましょう。`tests` ディレクトリ内にある `.spec.js` でファイル名が終わるものが、自動的に実行されます。全てがうまくいっていれば、以下のように画面に表示されます。
+
+Run the test by typing `yarn test:unit` into your terminal. Any file in the `tests` directory ending with `.spec.js` is automatically executed. If everything went well, you should see:
+
+```
+PASS  tests/unit/Greeting.spec.js
+Greeting.vue
+  ✓ renders a greeting (27ms)
+
+console.log tests/unit/Greeting.spec.js:7
+  <div>
+    Vue and TDD
+  </div>
+```
+
+先ほど書いたマークアップが正しく、テストが通ったということですね。ただし、このテストはいつでも通ります。なぜなら assertion をしていないので絶対に失敗しないからです。(それでももちろん有用ではあるのですが。) たとえ `Greeting.vue` を書き換えて `{{ message }}` を削除してしまっても、このテストはパスしてしまいます。ではこれに変更を加えていきましょう。
+
+We can see the markup is correct, and the test passes. The test is passing because there was no failure - this test can never fail, so it is very useful yet. Even if we change `Greeting.vue` and delete the `{{ message }}`, it will still pass. Let's change that.
+
+## Assertion を作成する
+
+コンポーネントが期待する挙動を正しくしているかどうかを判断するためには、assertion を作成する必要があります。そのためには Jest の `expect` API を使用します。こんなふうにです。`expect(result).to [matcher] (actual)`
+
+We need to make an assertion to ensure the component is behaving correctly. We can do that using Jest's `epxect` API. It looks like this: `expect(result).to [matcher] (actual)`. 
+
+_matcher_ の部分には、値やオブジェクトと比較するためのメソッドが使用されます。例えば次のようにです。 
+
+_Matchers_ are methods to compare values and objects. For example:
+
+```js
+expect(1).toBe(1)
+```
+
+全ての matcher は [Jest documentation](http://jestjs.io/docs/en/expect) で見ることができます。`vue-test-utils` ライブラリには一切 matcher は含まれていません。Jest が提供する matcher が十分なものであるからです。`Greeting` の中のテキストを比較したいので次のように書いてみましょう。
```

Comment:
Jest has official Japanese documentation now! https://jestjs.io/ja/

Can you include links to the Japanese documentation?

Reply:
OK!!